### PR TITLE
fix: naming convention for `UInt` lemmas

### LINKED ISSUE #4513: Some uint lemmas do not follow correct naming conventions
### Prerequisites

Please put an X between the brackets as you perform the following steps:

* [x] Check that your issue is not already filed:
      https://github.com/leanprover/lean4/issues
* [ ] Reduce the issue to a minimal, self-contained, reproducible test case.
      Avoid dependencies to Mathlib or Batteries.
* [ ] Test your test case against the latest nightly release, for example on
      https://live.lean-lang.org/#project=lean-nightly
      (You can also use the settings there to switch to “Lean nightly”)

### Description

The `UInt` lemmas `zero_toNat`, `div_toNat`, `mod_toNat`, `modn_toNat` reverse the normal convention where `toNat` should come first.

### Impact

Add :+1: to [issues you consider important](https://github.com/leanprover/lean4/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc). If others are impacted by this issue, please ask them to add :+1: to it.

### PR DESCRIPTION
**Pull request by fgdorais**
Closes #4513

### CODE REVIEW

File: src/Init/Data/UInt/Lemmas.lean [hunk 1]
```diff
@@ -45,10 +45,10 @@ protected theorem ne_of_val_ne {a b : $typeName} (h : a.val ≠ b.val) : a ≠ b
 open $typeName (ne_of_val_ne) in
 protected theorem ne_of_lt {a b : $typeName} (h : a < b) : a ≠ b := ne_of_val_ne (Fin.ne_of_lt h)
 
-@[simp] protected theorem zero_toNat : (0 : $typeName).toNat = 0 := Nat.zero_mod _
-@[simp] protected theorem mod_toNat (a b : $typeName) : (a % b).toNat = a.toNat % b.toNat := Fin.mod_val ..
-@[simp] protected theorem div_toNat (a b : $typeName) : (a / b).toNat = a.toNat / b.toNat := Fin.div_val ..
-@[simp] protected theorem modn_toNat (a : $typeName) (b : Nat) : (a.modn b).toNat = a.toNat % b := Fin.modn_val ..
+@[simp] protected theorem toNat_zero : (0 : $typeName).toNat = 0 := Nat.zero_mod _
+@[simp] protected theorem toNat_mod (a b : $typeName) : (a % b).toNat = a.toNat % b.toNat := Fin.mod_val ..
```
Comment (eric-wieser) [inline src/Init/Data/UInt/Lemmas.lean:49]: You could argue that `Fin.mod_val` is backwards too, and that this change now makes them inconsistent.

Comment (fgdorais) [reply src/Init/Data/UInt/Lemmas.lean:49]: I agree but I think that's another PR since these are actually used (e.g. here).

**Comment by semorrison** [comment]
@fgdorais could you please add `@[deprecated (since := "2024-06-23")] abbrev foo := @bar` for these? Happy to merge after that.

**Comment by fgdorais** [comment]
Deprecation completed. I kept these out of the theorem generating macro since I suppose they should eventually be removed. Let me know if you would rather have the deprecations inside the macro.

**Pull request merged**